src/Facility.CodeGen.JavaScript/JavaScriptGenerator.cs

```diff
@@ -278,7 +279,9 @@ public override CodeGenOutput GenerateOutput(ServiceInfo service)
 									var statusCodeAsString = ((int) validResponse.StatusCode).ToString(CultureInfo.InvariantCulture);
 
 									var bodyField = validResponse.BodyField;
-									var shouldExpectJsonResult = bodyField == null || service.GetFieldType(bodyField.ServiceField)!.Kind != ServiceTypeKind.Boolean;
+									var shouldExpectJsonResult =
+											(bodyField == null || service.GetFieldType(bodyField.ServiceField)!.Kind != ServiceTypeKind.Boolean) &&
+											validResponse.StatusCode != HttpStatusCode.NoContent;
```

Comment:
`HttpStatusCode.NotModified` is also an empty-content status code.

But it might be simpler/better to just match the conditional logic below and only check for `result.json` when it is used, i.e. `validResponse.NormalFields!.Count != 0` (when `bodyField == null`).